internal/expr/bindtypes.go

```diff
@@ -142,26 +142,33 @@ func (e *asteriskInsertExpr) bindTypes(argInfo typeinfo.ArgInfo) (tie any, err e
 		}
 	}()
 
-	var inputs []typeinfo.Input
-	var columns []string
+	var cols []insertColumn
 	for _, source := range e.sources {
 		if source.memberName == "*" {
-			inps, tags, err := argInfo.AllStructInputs(source.typeName)
+			inputs, tags, err := argInfo.AllStructInputs(source.typeName)
 			if err != nil {
 				return nil, err
 			}
-			columns = append(columns, tags...)
-			inputs = append(inputs, inps...)
+			for i, input := range inputs {
+				cols = append(cols, insertColumn{
```

Comment:
Worth a constructor that makes this `insertColumn`, then these are nice one-line functions.

Reply:
Added the constructor. This now also mirrors the `newOutputColumn` constructor.